adds `aria-{invalid,describedby}` annotations for WTForms validation errors

cfm opened the pull request:
## Status

Ready for review

## Description of Changes

Closes #6238 (towards #5972) by adding `aria-invalid` and `aria-describedby` annotations where possible for validated WTForms-based forms.  This is admittedly a pretty minor improvement; see #6238 for details on all the places where even this improvement is either not applicable or not possible in the current interfaces of the Source and Journalist Interfaces.

## Testing

1. Log in to the Journalist Interface as an administrator.
2. Begin adding a user and enter invalid values for each field.
3. Click **Add User**.
4. When the form comes back with validation errors, view source and observe that each invalid field is marked `aria-invalid` and has an `aria-describedby` annotation pointing to its validation errors.  (These annotations will also be used by assistive tech.)

## Deployment

No deployment considerations.

## Checklist

### If you made changes to the server application code:

- [x] Linting (`make lint`) and tests (`make test`) pass in the development container

cfm committed:
adds ARIA-{INVALID,DESCRIBEDBY} annotations for WTForms validation errors

legoktm commented on securedrop/journalist_templates/admin_add_user.html:
```diff
@@ -11,10 +11,16 @@ <h1 id="add-user-heading">Add User</h1>
+        {% endif %}
+        {{ form.username(**aria_attributes) }}
         {% for error in form.username.errors %}
+        <span id="username-errors">
```
Doesn't this need to be outside the {% for ... %} loop?

cfm replied:
Thanks for catching this!  Fixed (here and below) in b717eee2f4b751ea767610c1492be5b50c421b6a.

legoktm commented on securedrop/journalist_templates/admin_add_user.html:
```diff
@@ -11,10 +11,16 @@ <h1 id="add-user-heading">Add User</h1>
-        {{ form.username() }}
+        {% set aria_attributes = {} %}
+        {% if form.username.errors %}
+        {% set _dummy = aria_attributes.update({'aria-describedby': 'username-errors', 'aria-invalid': 'true'}) %}
```
jinja has a `do` expression that lets you avoid the set _dummy: https://jinja.palletsprojects.com/en/3.0.x/templates/#expression-statement

If we end up using this pattern elsewhere I think it would be nice to enable that extension.

cfm replied:
I like this suggestion.  We only use this pattern in one other template, in the Source Interface, but I've gone ahead and tacked on this quick refactoring in d72e0dc724f8d85e099215f34ac8f1f2d4afe94c.

legoktm commented on securedrop/journalist_templates/admin_add_user.html:
```diff
@@ -46,20 +64,32 @@ <h1 id="add-user-heading">Add User</h1>
+      {{ form.is_admin(id="is-admin", **aria_attributes) }}
       <label for="is-admin">{{ gettext('Is Admin') }}</label>
       {% for error in form.is_admin.errors %}
+      <span id="is-admin-errors">
```
Should be outside loop.

cfm commented:
Thanks for reviewing, @legoktm.  I'll revise this week.

cfm committed:
refactors "set _dummy" expressions with jina2.ext.do

thread: https://github.com/freedomofpress/securedrop/pull/6240#discussion_r810320511

cfm commented:
Thanks for the review, @legoktm.  Revised accordingly, including the suggested minor refactoring to Jinja's `do` expression.

legoktm reviewed (approved):
LGTM, test plan still checks out.
